cmd/operator-sdk/internal/genutil/crds.go

```diff
@@ -31,49 +28,12 @@ import (
 func CRDGen() error {
 	projutil.MustInProjectRoot()
 
-	absProjectPath := projutil.MustGetwd()
-	repoPkg := projutil.GetGoPkg()
-
-	gvMap, err := k8sutil.ParseGroupSubpackages(scaffold.ApisDir)
-	if err != nil {
-		return fmt.Errorf("failed to parse group versions: (%v)", err)
-	}
-	gvb := &strings.Builder{}
-	for g, vs := range gvMap {
-		gvb.WriteString(fmt.Sprintf("%s:%v, ", g, vs))
-	}
+	log.Info("Running CRD generator.")
 
-	log.Infof("Running CRD generation for Custom Resource group versions: [%v]\n", gvb.String())
-
-	s := &scaffold.Scaffold{}
-	cfg := &input.Config{
-		Repo:           repoPkg,
-		AbsProjectPath: absProjectPath,
-		ProjectName:    filepath.Base(absProjectPath),
-	}
-	crds, err := k8sutil.GetCRDs(scaffold.CRDsDir)
-	if err != nil {
-		return err
-	}
-	for _, crd := range crds {
-		g, v, k := crd.Spec.Group, crd.Spec.Version, crd.Spec.Names.Kind
-		if v == "" {
-			if len(crd.Spec.Versions) != 0 {
-				v = crd.Spec.Versions[0].Name
-			} else {
-				return fmt.Errorf("crd of group %s kind %s has no version", g, k)
-			}
-		}
-		r, err := scaffold.NewResource(g+"/"+v, k)
-		if err != nil {
-			return err
-		}
-		err = s.Execute(cfg,
-			&scaffold.CRD{Resource: r, IsOperatorGo: projutil.IsOperatorGo()},
-		)
-		if err != nil {
-			return err
-		}
+	cfg := genutil.Config{InputDir: scaffold.ApisDir, OutputDir: scaffold.CRDsDir}
+	crd := gencrd.NewCRDGo(cfg)
+	if err := crd.Generate(); err != nil {
+		return errors.Wrapf(err, "error generating CRDs from APIs in %s", scaffold.ApisDir)
```

Comment:
Can we start switching over to Go 1.13 errors?
```suggestion
		return fmt.Errorf("error generating CRDs from APIs in %s: %w", scaffold.ApisDir, err)
```

Reply:
`errors.Wrap` returns an error implementing Go 1.13’s `Unwrap` function. However I am fine with using `fmt.Errorf` everywhere for consistency.

Reply:
Oh that's cool, didn't know that. Regardless, if we can move to `fmt.Errorf` that's one less dependency to pull in. We can standardize in a follow-on.